### PR TITLE
test: Add e2e tests to cover additional scenarios where external files are rendered

### PR DESCRIPTION
**Pull request by Camila-B**
* Adds tests that cover rendering outside of the root directory: a local valuesFile, a local file that is referenced in a `configMapGenerator`, and a local file referenced in the `resources` field
* b/355278028

### CODE REVIEW

File: e2e/nomostest/testpredicates/predicates.go [hunk 1]
```diff
@@ -1518,3 +1518,20 @@ func RootSyncHasNomosStatus(expectedCommit, expectedStatus string) Predicate {
...
+func ConfigMapHasData(key string, value string) Predicate {
+	return func(o client.Object) error {
+		if o == nil {
+			return ErrObjectNotFound
+		}
+		cm, ok := o.(*corev1.ConfigMap)
+		if !ok {
+			return WrongTypeErr(o, cm)
+		}
+		if cm.Data[key] == value {
```
Comment (sdowell) [inline e2e/nomostest/testpredicates/predicates.go:1532]: Suggest checking that the value exists as well. Currently this does not differentiate between the key missing and the key containing an empty string.

```golang
if gotVal, ok := cm.Data[key]; !ok {
  return fmt.Errorf(...)
} else if gotVal != value {
  return fmt.Errorf(...)
}
return nil
```

File: e2e/testcases/hydration_test.go [hunk 2]
```diff
@@ -170,6 +170,42 @@ func TestHydrateKustomizeComponents(t *testing.T) {
...
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("add DRY configs to the repository"))
+
+	nt.T.Log("Update RootSync to sync from the external-files directory")
+	rs := k8sobjects.RootSyncObjectV1Beta1(configsync.RootSyncName)
+	nt.MustMergePatch(rs, `{"spec": {"git": {"dir": "external-files"}}}`)
+	nt.Must(nt.WatchForAllSyncs(nomostest.WithSyncDirectoryMap(syncDirMap)))
+
+	// Validate nomos status
+	latestCommit := nt.RootRepos[configsync.RootSyncName].MustHash(nt.T)
+	nt.Must(nt.Validate(configsync.RootSyncName, configsync.ControllerNamespace, &v1beta1.RootSync{},
```
Comment (sdowell) [inline e2e/testcases/hydration_test.go:194]: I think this check is redundant with the `WatchForAllSyncs` check above. Is there something specific you're trying to assert here?

Comment (Camila-B) [reply e2e/testcases/hydration_test.go]: No I was just copying the general structure of the other tests which have a similar check. I went ahead and removed it.

File: e2e/nomostest/testpredicates/predicates.go (hunk 1 above)
Comment (Camila-B) [reply e2e/nomostest/testpredicates/predicates.go]: Thanks, done

File: e2e/testdata/hydration/external-files/namespace.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2022 Google LLC
```
Comment (sdowell) [inline e2e/testdata/hydration/external-files/namespace.yaml:1]: 2024

Comment (Camila-B) [reply e2e/testdata/hydration/external-files/namespace.yaml]: Thanks for catching that! Done

**Review by sdowell — APPROVED** [review]
/lgtm

**Pull request merged**